### PR TITLE
Lazy search tabs

### PR DESCRIPTION
**Pull request by vevix**
:arrow_up: :arrow_down:

Also debounces search so it isn't as jumpy.

### CODE REVIEW

File: src/screens/Search/SearchScreen/index.js [hunk 1]
```diff
@@ -30,55 +30,43 @@ class SearchScreen extends Component {
...
-  handleSearchStateChange = (route, query) => {
-    // const { query } = searchState;
-    const nextQueryState = { ...this.state.query, [route.key]: query !== '' ? query : undefined };
-    this.setState({ query: nextQueryState }, () => {
-      this.doSearch(query, route);
-    });
-  };
-
-  handleIndexChange = index => this.setState({ index });
+  debouncedSearch = debounce(this.executeSearch, 250);
```
Comment (NuckChorris) [inline src/screens/Search/SearchScreen/index.js]: I'd lower this to 100-200ms, honestly. Algolia is quick enough that we don't need as much debouncing and we make the search feel quicker by reducing this

File: src/screens/Search/SearchScreen/index.js [hunk 2]
```diff
@@ -30,55 +30,43 @@ class SearchScreen extends Component {
...
+   const nextState = { ...this.state.query, [scene]: isEmpty(query) ? undefined : query };
+   this.setState({ query: nextState }, () => {
+     this.debouncedSearch(query, scene);
+   });
+  };
 
-    algoliaIndex.search({ query, filters }, (err, content) => {
+  executeSearch = (query, scene) => {
+    const currentScene = this.state.scenes[scene];
+    const client = algolia(kitsuConfig.algoliaAppId, currentScene.apiKey);
```
Comment (Mikunj) [inline src/screens/Search/SearchScreen/index.js]: There could be a case here where `currentScene.apiKey` is null, and if so algolia won't be initialized, maybe just return early?

Comment (vevix) [reply src/screens/Search/SearchScreen/index.js]: Yeah, that sucks :(

Comment (vevix) [reply src/screens/Search/SearchScreen/index.js]: Fixed, we should maybe flip a flag there and show a failed state.

**Pull request merged**